core/os/device/device.proto

```diff
@@ -269,3 +271,20 @@ message VulkanPhysicalDevice {
   // deviceName is a null-terminated string containing the name of the device.
   string device_name = 5;
 }
+
+// DeviceValidationCache contains the list of validated devices.
+// The cached information is stored into a file whenever a new device is
+// successfully validated.
+message DeviceValidationCache {
+  repeated DeviceValidationCacheEntry entries = 1;
+}
+
+// DeviceValidationCacheEntry describes the properties of the device that can
+// change validation results if updated.
+message DeviceValidationCacheEntry {
+  OS os = 1;
+  // driverVersion from the VulkanPhysicalDevice. (Per VulkanPhysicalDevice)
+  repeated uint32 driver_version = 2;
+  // The pre-release driver APK version code
+  int64 version_code = 3;
+}
```

Comment:
The current idea is that we cache only the devices that passed validation. If we cache even the failed ones, then we need to enable a way to re-run validation, which I feel is going to be an additional button or something and complicates stuff.
Given that if validation fails on one driver version, it is never going to pass on that driver again. But if validation failed because the user mistakenly presses the home button or does something with the device, then the only way to re-run validation in the current state would be to delete the cache and try again. To avoid this situation, I feel its better to just not cache failed devices. 
What do you think @lpy and @pmuetschard ?

Reply:
OK. Let's design for caching success and failure (i.e. the proto message), but for v1 only actually cache successes.